packages/hub/node-tests/services/web3-socket-test.ts

```diff
@@ -2,30 +2,115 @@ import Web3SocketService from '../../services/web3-socket';
 import sentryTestkit from 'sentry-testkit';
 import * as Sentry from '@sentry/node';
 import waitFor from '../utils/wait-for';
+import { server as WebsocketServer } from 'websocket';
+import http from 'http';
 
 const { testkit, sentryTransport } = sentryTestkit();
 const DUMMY_DSN = 'https://acacaeaccacacacabcaacdacdacadaca@sentry.io/000001';
 
 describe('Web3Socket', function () {
-  it('reports to sentry when web3 initialization fails', async function () {
+  let subject: Web3SocketService;
+  let wsServer: WebsocketServer;
+  let httpServer: http.Server;
+
+  this.beforeEach(function () {
+    httpServer = http.createServer(function (_, response) {
+      response.end();
+    });
+    httpServer.listen(8545);
+    wsServer = new WebsocketServer({
+      httpServer,
+      autoAcceptConnections: true,
+    });
+    wsServer.on('request', function (request: any) {
+      request.accept('echo-protocol', request.origin);
+    });
+    subject = new Web3SocketService();
+    subject.rpcURL = 'ws://localhost:8545';
```

Comment:
is there any possibility this port will be taken? I’ve sometimes seen test suites ask for a random unassigned port vs hard-coding one 🤔

Reply:
Oh, that might be good! though I'm not very sure how to do that. I picked this because it's the default hardhat (private blockchain, with js implementations: https://hardhat.org/) ws url.